canonical/maas-ui
# feat: Create new Security settings section in settings

## Done

- Removed "Security" from configuration settings
- Created new "Security" section in settings
- Moved Vault settings to "Secret storage" under security
- Moved TLS settings to "Security protocols" under security
- Moved IPMI section of commissioning settings form from configuration/commissioning to security/ipmi-settings
- Adapted old tests to not check for now missing parts of pages
- Created new tests for new pages and components

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to /settings
- Verify that a new "Security" nav item is shown
- Verify that clicking on teach subnav item under this loads the page it describes
- Navigate to /settings/security/ipmi-settings
- Change some of the form values and submit
- Verify that the form submission works 
- Verify that the new data is shown upon page refresh
- Navigate to /settings/configuration/commissioning
- Change some of the form values and submit
- Verify that the form submission works 
- Verify that the new data is shown upon page refresh
- Navigate to /settings/configuration/security
- Verify that a "Not found" error is shown

## Screenshots

### Before 

`/settings/configuration/security`
![image](https://user-images.githubusercontent.com/35104482/207659004-b192fae5-8c3f-4dbd-b0e9-1bcd191dcb43.png)

`settings/configuration/commissioning`
![image](https://user-images.githubusercontent.com/35104482/207659456-0f0ad3ad-2844-4305-956d-8ffcc0914e57.png)

### After
`/settings/security/security-protocols`
![image](https://user-images.githubusercontent.com/35104482/207659691-10b22264-7cd4-47b2-b1b5-8c661e316233.png)

`/settings/security/secret-storage`
![image](https://user-images.githubusercontent.com/35104482/207659765-779d8fb0-c5bf-48e2-9689-f512dd210636.png)

`/settings/security/ipmi-settings`
![image](https://user-images.githubusercontent.com/35104482/207659839-426018c1-fc74-48dc-a4f4-d178869a1188.png)

`/settings/configuration/commissioning`
![image](https://user-images.githubusercontent.com/35104482/207659915-68c13cc5-843a-4c0b-984b-fb944884237b.png)
